ruby/sdk-phase-1.md

```diff
@@ -0,0 +1,496 @@
+  * 💭 Why? Users wanting to simply set JSON options shouldn't have to go through the mess of recreating the default
+    list of encoding converters.
+
+## Activities
+
+A simple, heartbeating activity that calls a separate thing:
+
+```ruby
+class MyActivity < Temporalio::Activity
+  # Can customize name, e.g.
+  # activity_name 'custom-name'
+
+  # Can customize executor, e.g.
+  # activity_executor :fiber
+
+  # Can make an activity dynamic, e.g.
+  # activity_dynamic true
+
+  def initialize(db_client)
+    @db_client = db_client
+  end
+
+  def execute(arg)
+    loop do
+      db_result = @db_client.do_thing(arg)
+      if db_result
+        return db_result
+      end
+      sleep 1
+      Temporalio::Activity::Context.heartbeat
+    end
+  end
+end
+```
+
+* Activities are defined as classes.
+  * 💭 Why classes instead of methods/functions like every other SDK? There are tradeoffs to both approaches. As
+    classes, they are more easily referenced, a bit more easily typed in RBS, and metadata can be applied more easily at
+    the class level. This comes at a cost of not being able to easily share state across classes, but users can provide
+    something in the constructor multiple activities use. Open to considering activity methods instead.
+  * ❓ Would we prefer activities as methods knowing they may not be able to be typed/referenced well? We can't really
+    take the workflow signal-ish approach of those pseudo-decorators making class methods unless we say activities must
+    be instance methods (since we would allow class methods as activities).
+* If an activity class is provided to a worker, it is instantiated each attempt. But if an activity instance is provided
+  to a worker, execute is called on that instance.
+  * 💭 Why support every-attempt-instantiated activities? This is a reasonable expectation for Ruby things and is how
+    the current Temporal Ruby SDK operates. This allows users to use instance state without race concerns.
+  * 💭 Why support pre-instantiated activities? It is a very common need for SDKs to be able to support cross-attempt
+    state such as a DB client, and while Ruby users are unfortunately used to class-level/static attrs, it should not
+    be encouraged.
+  * ❓ Is it too confusing that the instance lifecycle is different depending on whether you pass in a class or an
+    instance?
+* `Temporalio::Activity::Context` is a thread/fiber-local context available to activities.
+  * It has instance methods for info, heartbeat, etc.
+  * It has a class method for `current` that returns from thread/fiber-local and class method shortcuts for all instance
+    methods.
+    * 💭 Why class methods equivalents? `Temporalio::Activity::Context.heartbeat` is easier to read than
+      `Temporalio::Activity::Context.current.heartbeat`. In .NET we made them get current instance first because that's
+      normal there, and in Python we only offered package-level methods because that's normal there.
```

Comment:
```suggestion
  * It has a class method for `current` that returns from thread/fiber-local.
```

I think we are going to walk back this decision for also having class methods. I think it's not really common in Ruby to define class and instance methods of the same name, and `Singleton` and others show that people are ok calling `.current` or something.